src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Scoring;
+using Microsoft.ML.OnnxScoring;
+using OnnxShape = System.Collections.Generic.List<long>;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    typeof(OnnxTransform.Arguments), typeof(SignatureDataTransform), OnnxTransform.UserName, OnnxTransform.ShortName)]
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    null, typeof(SignatureLoadDataTransform), OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(OnnxTransform), null, typeof(SignatureLoadModel),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(OnnxTransform), null, typeof(SignatureLoadRowMapper),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(OnnxTransform))]
+
+namespace Microsoft.ML.OnnxScoring
+{
+    public sealed class OnnxTransform : ITransformer, ICanSaveModel
+    {
+        public sealed class Arguments : TransformInputBase
```

Comment:
>Arguments [](start = 28, length = 9)

So far we have tendency to use Arguments class for entry points and command line, and to have separate class ColumnInfo with same functionality, but without all attributes.
And in our constructors we tend to use ColumnInfo class (except constructor for SignatureLoadDataTransform)

Reply:
I was basing this on TensorFlowTransform which only has Arguments class. I looked at couple  other transforms (MutualInformationFeatureSelectionTransform, GroupTransform) and they have Arguments only. Could you please point me to a sample transform that follows above?

Reply:
[Adding Shahab's response]
I was basing this on TensorFlowTransform which only has Arguments class. I looked at couple  other transforms (MutualInformationFeatureSelectionTransform, GroupTransform) and they have Arguments only. Could you please point me to a sample transform that follows above?

---
In reply to: [219613034](https://github.com/dotnet/machinelearning/pull/942#discussion_r219613034) [](ancestors = 219613034)